src/VirtualMonoRepo/eng/src/VmrSyncJobTemplate.cs

```diff
@@ -0,0 +1,127 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Sharpliner;
+using Sharpliner.AzureDevOps;
+using Sharpliner.AzureDevOps.ConditionedExpressions;
+
+namespace Pipeline;
+
+internal class VmrSyncJobTemplate : JobTemplateDefinition
+{
+    #region YAML export settings
+
+    public const string TemplatePath = "common/templates/jobs/vmr-synchronization.yml";
+    public override TargetPathType TargetPathType => TargetPathType.RelativeToGitRoot;
+    public override string TargetFile => $"{Configuration.PipelineRootDir}/{TemplatePath}";
+
+    #endregion
+
+    #region Environment definitions
+
+    private static string InstallerPath => $"{variables.Agent.BuildDirectory}/installer";
+    private static string VmrPath => $"{variables.Agent.BuildDirectory}/vmr";
+
+    #endregion
+    
+    public override List<Parameter> Parameters { get; } = new()
+    {
+        StringParameter("targetRef", "Target revision in dotnet/installer to synchronize", variables.Build.SourceVersion),
+        StringParameter("vmrBranch", $"{Configuration.VmrName} branch to use", variables.Build.SourceBranchName),
+    };
+
+    public override string[]? Header => base.Header!.Concat(new[]
+    {
+        $"These steps synchronize code from product repositories into the VMR (https://github.com/{Configuration.VmrName})"
+    }).ToArray();
+
+    public override ConditionedList<JobBase> Definition => new()
+    {
+        new Job("Synchronize", $"Synchronize {Configuration.VmrName}")
+        {
+            Pool =
+                If.Equal(variables.System.TeamProject, "public")
+                    .Pool(new HostedPool()
+                    {
+                        Name = "NetCore-Public",
+                        Demands = new() { "ImageOverride -equals Build.Ubuntu.2004.Amd64.Open" }
+                    })
+                .Else
+                    .Pool(new HostedPool()
+                    {
+                        Name = "NetCore1ESPool-Internal",
+                        Demands = new() { "ImageOverride -equals Build.Ubuntu.2004.Amd64" }
+                    }),
+
+            Timeout = TimeSpan.FromHours(2),
+            Steps =
+            {
+                Checkout.Self with
+                {
+                    DisplayName = "Checkout dotnet/installer",
+                    Clean = true,
+                    Path = "installer",
+                    FetchDepth = 0,
+                },
+
+                Checkout.Repository("vmr") with
+                {
+                    DisplayName = $"Checkout {Configuration.VmrName}",
+                    Timeout = TimeSpan.FromMinutes(30),
+                    Clean = true,
+                    Path = "vmr",
+                    FetchDepth = 0,
```

Comment:
Ah, this probably needs to be `1`

```suggestion
                    FetchDepth = 1,
```